OSDOCS-11208-storage-nodes: Documented node and storage bug fixes 4.17 (#82531)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

[id="ocp-4-17-node-bug-fixes_{context}"]
==== Node

> * Previously, the Container Runtime Config controller did not consider whether a mirror configuration was in use when adding the scope from a `ClusterImagePolicy` CR to /etc/containers/registries.d/sigstore-registries.yaml. As a consequence, image verification failed with a `Not looking for sigstore attachments` message. With this fix, images are pulled from the mirror registry as expected. (link:https://issues.redhat.com/browse/OCPBUGS-36344[*OCPBUGS-36344*])

Reviewer: ```suggestion
* Previously, the Container Runtime Config controller did not detect whether a mirror configuration was in use before adding the scope from a `ClusterImagePolicy` CR to the `/etc/containers/registries.d/sigstore-registries.yaml` file. As a consequence, image verification failed with a `Not looking for sigstore attachments` message. With this fix, images are pulled from the mirror registry as expected. (link:https://issues.redhat.com/browse/OCPBUGS-36344[*OCPBUGS-36344*])
```

Author: :heavy_check_mark:

> * Previously, a group ID was not added to the /etc/group within a container when the `spec.securityContext.runAsGroup` attribute was set in the pod spec. With this release, this issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-39478[*OCPBUGS-39478*])

Reviewer: ```suggestion
* Previously, a group ID was not added to the `/etc/group` directory within a container when the `spec.securityContext.runAsGroup` attribute was set in the pod specification. With this release, this issue is fixed. (link:https://issues.redhat.com/browse/OCPBUGS-39478[*OCPBUGS-39478*])
```
Hm, do we use "spec" or "specification" in plain language? Specification is probably better, but I don't see a rule.

Author: :heavy_check_mark:

* Previously, because of a critical regression on RHEL 9.4 kernels earlier than `5.14.0-427.26.1.el9_4`, the `mglru` feature had memory management disabled. In this release, the regression issue is fixed so that the `mglru` feature is now enabled in {product-title} {product-version}. (link:https://issues.redhat.com/browse/OCPBUGS-35436[*OCPBUGS-35436*])

[discrete]
[id="ocp-4-17-node-tuning-operator-bug-fixes_{context}"]
==== Node Tuning Operator (NTO)

> * Previously, due to an internal bug, the Node Tuning Operator incorrectly computed CPU masks for interrupt and network handling CPU affinity if a machine had more than 256 CPUs. This prevented proper CPU isolation on those machines and resulted in `systemd` unit failures. With this release, the Node Tuning Operator computes the masks correctly. (link:https://issues.redhat.com/browse/OCPBUGS-39164[*OCPBUGS-39164*]) 

Reviewer: ```suggestion
* Previously, due to an internal bug, the Node Tuning Operator incorrectly computed CPU masks for interrupt and network-handling CPU affinity if a machine had more than 256 CPUs. This prevented proper CPU isolation on those machines and resulted in systemd unit failures. With this release, the Node Tuning Operator computes the masks correctly. (link:https://issues.redhat.com/browse/OCPBUGS-39164[*OCPBUGS-39164*]) 
```
We don't backtick systemd in MicroShift because RHEL doesn't, see https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/using_systemd_unit_files_to_customize_and_optimize_your_system/index. Doesn't look like OCP does, either, see https://docs.openshift.com/container-platform/4.16/observability/logging/config/cluster-logging-systemd.html

Author: :checklist

[id="ocp-4-17-storage-bug-fixes_{context}"]
==== Storage

> * Previously, the Secrets Store Container Storage Interface (CSI) Driver on HyperShift hosted clusters was failing to mount secrets because of an issue when using the HyperShift CLI to create OpenID Connect (OIDC) infrastructure on {aws-full}. With this release, the issue has been fixed so that the driver can now mount volumes. (link:https://issues.redhat.com/browse/OCPBUGS-18711[*OCPBUGS-18711*])

Reviewer: ```suggestion
* Previously, the Secrets Store Container Storage Interface (CSI) Driver on Hosted Control Plane clusters failed to mount secrets because of an issue when using the CLI to create OpenID Connect (OIDC) infrastructure on {aws-full}. With this release, the issue has been fixed so that the driver can now mount volumes. (link:https://issues.redhat.com/browse/OCPBUGS-18711[*OCPBUGS-18711*])
```
Hypershift is internal; product is Hosted Control Planes. I believe HCPs uses the ROSA CLI? Asked @lahinson

Author: :heavy_check_mark:
